From 9ca8616e6fba5257f7678c0283e37d757832fa46 Mon Sep 17 00:00:00 2001
From: IgnorantGuru <ignorantguru@gmx.com>
Date: Tue, 22 Mar 2016 06:14:51 -0600
Subject: [PATCH] [installer] fix mktemp template and error handling for
 busybox #629

fixes mktemp template for busybox compatibility - no suffix allowed in
busybox without explicity --suffix ?

CRITICAL If the mktemp command fails, installer may remove files in the
current dir due to incorrect error handling..
---
 ChangeLog         |  1 +
 spacefm-installer | 17 +++++++++++------
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index aae579c..29cceb6 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,6 @@
 1.0.5+next
     Rename dlg canonicalize Copy Target/Link Target target when possible
+    [installer] fix mktemp template and error handling for busybox #629
 1.0.5   2016-01-20:
     recreate too small cached thumbnails #584
     read $XDG_TEMPLATES_DIR via glib for user-dirs.dirs support #581
diff --git a/spacefm-installer b/spacefm-installer
index b483aea..cd79e75 100755
--- a/spacefm-installer
+++ b/spacefm-installer
@@ -182,14 +182,18 @@ if [ -n "$branch" ] || [ ! -e configure ]; then
     # Make temp dir
     if [ ! -w . ]; then
         # no write access in current dir so tell mktemp to use $TMPDIR
-        echo ">>> mktemp -dt spacefm-$branch-XXXXXXXX.tmp"
-        tdir="$(mktemp -dt spacefm-$branch-XXXXXXXX.tmp)"
+        echo ">>> mktemp -dt spacefm-build-$branch-XXXXXXXX"
+        tdir="$(mktemp -dt spacefm-build-$branch-XXXXXXXX)"
+        err=$?
     else
-        echo ">>> mktemp -d spacefm-$branch-XXXXXXXX.tmp"
-        tdir="$(pwd)/$(mktemp -d spacefm-$branch-XXXXXXXX.tmp)"
+        echo ">>> mktemp -d spacefm-build-$branch-XXXXXXXX"
+        tdir="$(mktemp -d spacefm-build-$branch-XXXXXXXX)"
+        err=$?
+        if [ "$tdir" != "" ]; then
+            tdir="$(pwd)/$tdir"
+        fi
     fi
-    source_archive="spacefm-$branch.tar.gz"
-    if [ -z "$tdir" ] || [ ! -d "$tdir" ]; then
+    if [ $err -ne 0 ] || [ -z "$tdir" ] || [ ! -d "$tdir" ]; then
         echo "spacefm-installer: creation of temporary dir failed" 1>&2
         exit 1
     fi
@@ -203,6 +207,7 @@ if [ -n "$branch" ] || [ ! -e configure ]; then
     fi
     
     # Download
+    source_archive="spacefm-$branch.tar.gz"
     download_url="${tarball_url}$branch.tar.gz"
     use_git=0
     if ( which wget &>/dev/null ); then
